pkg/controllers/dashboard/systemcharts/controller.go

```diff
@@ -548,6 +558,28 @@ func (h *handler) onPlan(_ string, plan *upgradev1.Plan) (*upgradev1.Plan, error
 	return plan, nil
 }
 
+// cleanUpEmbeddedCAPIWebhooks waits for the namespace created by the rancher-provisioning-capi chart,
+// cattle-provisioning-capi-system, to be removed and deletes the associated webhook configurations
+// initially deployed by Rancher on startup. These webhook configurations and namespace
+// are specific to the rancher-provisioning-capi chart that is being replaced with the turtles chart.
+func (h *handler) cleanUpEmbeddedCAPIWebhooks(_ string, ns *kcorev1.Namespace) (*kcorev1.Namespace, error) {
+	if ns == nil || ns.Name != namespace.ProvisioningCAPINamespace || ns.DeletionTimestamp == nil {
+		return ns, nil
+	}
+
+	err := h.mutatingWebhookConfigurations.Delete(capiMutatingWebhookName, &metav1.DeleteOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return ns, err
```

Comment:
Even though we’re not modifying the namespace, it’s best to follow the established convention.

```suggestion
		return nil, err
```

also apply to line 577